src/js/_enqueues/lib/emoji-loader.js

```diff
@@ -2,65 +2,168 @@
  * @output wp-includes/js/wp-emoji-loader.js
  */
 
-( function( window, document, settings ) {
-	var src, ready, ii, tests;
+/**
+ * Emoji Settings as exported in PHP via _print_emoji_detection_script().
+ * @typedef WPEmojiSettings
+ * @type {object}
+ * @property {?object} source
+ * @property {?string} source.concatemoji
+ * @property {?string} source.twemoji
+ * @property {?string} source.wpemoji
+ */
+
+/**
+ * Support tests.
+ * @typedef SupportTests
+ * @type {object}
+ * @property {?boolean} flag
+ * @property {?boolean} emoji
+ */
+
+/**
+ * IIFE to detect emoji support and load Twemoji if needed.
+ *
+ * @param {Window} window
+ * @param {Document} document
+ * @param {WPEmojiSettings} settings
+ */
+( function wpEmojiLoader( window, document, settings ) {
+	if ( typeof Promise === 'undefined' ) {
+		return;
+	}
 
-	// Create a canvas element for testing native browser support of emoji.
-	var canvas = document.createElement( 'canvas' );
-	var context = canvas.getContext && canvas.getContext( '2d' );
+	var sessionStorageKey = 'wpEmojiSettingsSupports';
+	var tests = [ 'flag', 'emoji' ];
+
+	/**
+	 * Checks whether the browser supports offloading to a Worker.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @private
+	 *
+	 * @returns {boolean}
+	 */
+	function supportsWorkerOffloading() {
+		return (
+			typeof Worker !== 'undefined' &&
+			typeof OffscreenCanvas !== 'undefined' &&
+			typeof URL !== 'undefined' &&
+			URL.createObjectURL &&
+			typeof Blob !== 'undefined'
+		);
+	}
+
+	/**
+	 * Get supports from session.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @private
+	 *
+	 * @returns {?SupportTests} Supports or null if not set.
+	 */
+	function getSessionSupports() {
+		if (
+			typeof sessionStorage !== 'undefined' &&
+			sessionStorageKey in sessionStorage
+		) {
+			try {
+				var supports = JSON.parse(
+					sessionStorage.getItem( sessionStorageKey )
+				);
+				if ( typeof supports === 'object' ) {
+					return supports;
+				}
+			} catch ( e ) {}
+		}
+		return null;
+	}
+
+	/**
+	 * Persist the supports in session storage.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @private
+	 *
+	 * @param {SupportTests} supports Supports.
+	 */
+	function setSessionSupports( supports ) {
```

Comment:
I don't think that's necessary because presumably any change to the platform emoji support would require a reboot, and that would result in a new session, correct?

Reply:
Seems that may not be correct. If rebooting and browser tabs are restored, then the `sessionStorage` would need to persist.

Reply:
I've made this change in d694445739

Reply:
The data stored in `sessionStorage` is only considered if it isn't older than a day.